src/AllocationBoundsInference.cpp

```diff
@@ -128,8 +128,7 @@ class AllocationInference : public IRMutator2 {
         for (map<string, Function>::const_iterator iter = e.begin();
              iter != e.end(); ++iter) {
             Function f = iter->second;
-            if (f.has_extern_definition() &&
```

Comment:
Why this change?

Reply:
The original code removed the touch tracking done here for any extern with a proxy expression. There isn't anything anywhere else that gets the info from the proxy. Also I'm assuming this code block doesn't actually depend on the bounds themselves but just that there is a touch. I may have this wrong, but I'm fairly certain I needed to make this change to get storage folding to happen in at least one case when working with the hex-dma2 branch.